xbmc/platform/posix/utils/PosixInterfaceForCLog.cpp

```diff
@@ -83,21 +83,25 @@ void CPosixInterfaceForCLog::PrintDebugString(const std::string &debugString)
 #endif // _DEBUG
 }
 
-void CPosixInterfaceForCLog::GetCurrentLocalTime(int &hour, int &minute, int &second, double &milliseconds)
+void CPosixInterfaceForCLog::GetCurrentLocalTime(int &year, int &month, int &day, int &hour, int &minute, int &second, double &milliseconds)
 {
   struct tm localTime;
   struct timeval tv;
 
   if (gettimeofday(&tv, nullptr) != -1 && localtime_r(&tv.tv_sec, &localTime) != NULL)
   {
+    year   = localTime.tm_year + 1900;
+    month  = localTime.tm_mon + 1;
+    day    = localTime.tm_mday;
     hour   = localTime.tm_hour;
     minute = localTime.tm_min;
     second = localTime.tm_sec;
     milliseconds = static_cast<double>(tv.tv_usec) / 1000;
   }
   else
   {
-    hour = minute = second = 0;
+    year = month = hour = minute = second = 0;
```

Comment:
month 0 is invalid

Reply:
tm_mon [should be](http://pubs.opengroup.org/onlinepubs/7908799/xsh/time.h.html) month of year [0,11] ?

Reply:
sure, but this is the error path, which has nothing to do with that :-)

In the event of an error you would get `0-0-1` which is not a valid date. `0-1-1` is (even if it is very unlikely to hit this code in either case...).

Reply:
Seems thats not used with any other function, just String as fallback for log. For me any of those two isnt "valid" date, from user POV 0-0-0 is more error suggestive, but I let you guys decide, which is best approach (maybe with consideration bigger picture and date usage around). If you want to have 0-1-1 I am fine with it.

Reply:
I don't care either way tbh, I'm only pointing out an inconsistency because you set the day to 1 but everything else to 0. If you want to set everything to 0 that's also fine by me.

Reply:
done :)